manifests/init.pp

```diff
@@ -215,7 +215,7 @@
   Boolean $service_enable = true,
   Boolean $service_ensure = true,
   Integer[0] $content_service_worker_count = (2*min(8, $facts['processors']['count']) + 1),
-  Integer[0] $api_service_worker_count = 1,
+  Integer[0] $api_service_worker_count = (2*min(4, $facts['processors']['count']) + 1),
```

Comment:
wouldn't that default to 9 or 10  as a minimum (2* 4 + 1)?  i think we probably want to drop the 2*, and just have:

min(4, $facts['processors']['count'])

although maybe we should set a max on it too (but maybe thats another bug as content_service_worker_count doesn'th ave a max.  Maybe don't worry about that here

Reply:
This was inspired by the content worker count calculation, and that comes from this https://docs.gunicorn.org/en/stable/design.html#how-many-workers so for what our default is here, I am not sure what to apply either. But note this is a default and users can tune this to their needs if need be.

So I suppose the question is still, what do we think our default should be and should it be dynamic to allow for user system resources to scale this value as we typically associate system resources with scale of deployment.

Reply:
I think it should scale with the # of processors, i was more stating that i think 2x is probably too much and a more 1 to 1 scaling probably makes more sense

Reply:
I would like to raise the fact that such computations are often for "this machine just runs my service", which is not true for most of the pulpcore installs we're looking at. We even have another `gunicorn` cluster next to this one for the content service.

So yeah, I would also probably look at a lower starting value, where `min(4, $facts['processors']['count']) + 1` sounds reasonable?

Reply:
since ehelms is out, i opened a new version here:  https://github.com/theforeman/puppet-pulpcore/pull/231